spec/features/user/user_login_spec.rb

```diff
@@ -21,10 +25,46 @@
   end
 
   it "prevents an invalid user from logging in" do
-    visit login_path
-    fill_in("session_email", with: "paul@mail.com")
-    fill_in("session_password", with: "not paul")
-    click_button "Sign in"
+    visit root_path
+    within("#signInModal") do
+      fill_in("session_email", with: "paul@mail.com")
+      fill_in("session_password", with: "not paul")
+      click_button "Sign in"
+    end
     expect(page).to have_content("Invalid login.")
   end
+
+  context "when used through Twitter" do
+    it "allows the user to log in" do
+      visit root_path
+      within("#signInModal") do
+        expect(page).to have_selector(:link_or_button, "Sign in with Twitter")
+      end
+      user = OmniAuth.config.mock_auth[:twitter] = OmniAuth::AuthHash.new({
+        'provider' => 'twitter',
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.